pravega-k8-log-collection/pravega-k8s-log.sh

```diff
@@ -0,0 +1,34 @@
+#/usr/bin/env bash
+set -euo pipefail
+mkdir -p /tmp/logdir
+output_dir="/tmp/logdir"
+kubectl get po,svc,pvc --all-namespaces > $output_dir/kubectl-get-po-svc-pvc-all-namespace_output
+IFS=$'\n'
+for i in $(cat $output_dir/kubectl-get-po-svc-pvc-all-namespace_output | grep -i pravega)
+do
+echo $i
+if [[ $i == *"po/"* ]] || [[ $i == *"pod/"* ]]; then
+    if [ "`echo $i | awk '{print $4}'`" == "Running" ]; then
+    {
+        if [ `echo $i | awk '{print $5}'` != 0 ]; then     
+            kubectl -n `echo $i | awk '{print $1}'` logs `echo $i | awk '{print $2}'` -p > $output_dir/log_`echo $i | awk 'gsub(/\//,"_") {print $2}'`_previous.log
+        fi
+        kubectl -n `echo $i | awk '{print $1}'` logs `echo $i | awk '{print $2}'` > $output_dir/log_`echo $i | awk 'gsub(/\//,"_") {print $2}'`_recent.log
```

Comment:
If I understand correctly the semantics of `kubectl logs`, this is getting the `stdout`. Does it mean we are not getting the logback output of the Pravega components? Or is it the case that we are configuring `logback` to output to `stdout`? Could you clarify, please?

Reply:
We are taking the `stdout` of `kubectl logs` what ever it is posting with default parameters and creating `_recent` file, which contains recent state of the pods log. Similarly, if the pod restarted `kubectl logs` have a provision to see `previous` logs (if exist) and we are taking that in `_previous` file. This would be fine if we are looking for most recent cases for debug but not complete logging as explained above and hence complete log directory copy is being done if script detect there are many log files and that had been copied under `oldlogs_<pod-name>` directory.